### ISSUE TITLE
[Build] How to use cmake to compile and generate 'onnxruntime. dll' from source code in Windows

### ISSUE THREAD

**Issue by HzHzHzHzHz** [microsoft/onnxruntime#16739]
### Describe the issue

Compile from source code using cmake on Windows using default options to generate the following lib, but without 'onnxruntime. dll'“
![image](https://github.com/microsoft/onnxruntime/assets/49800264/80cccb20-24df-4295-beab-e0e739bf4547)

### Target platform

Windows

### Build script

Cmake 3.26.4+VS2019

### Error / output

no 'onnxruntime.dll' in the generated file 
![image](https://github.com/microsoft/onnxruntime/assets/49800264/25fca9d2-4b9e-4412-ac0f-8f80bed07c9d)

**Comment by xadupre** [comment]
Can you share the command line you used to build onnxruntime?

**Comment by HectorSVC** [comment]
try add "--build_shared_lib"

**Comment by HzHzHzHzHz** [comment]
> Can you share the command line you used to build onnxruntime?

It has been resolved. Need Cmake to open onnxruntime_ BUILD_ SHARED_ LIB configuration item.

**Comment by HzHzHzHzHz** [comment]
> try add "--build_shared_lib"
Perhaps this method can work in the official "bat" command line format. But I compiled the source code directly using Cmake. After my testing, opening the "build_shared_lib" configuration item in Cmake did not solve the problem, and it is also possible that there was a problem with my other operations. But after referring to your approach, the problem has been resolved.
Need Cmake to open onnxruntime_ BUILD_ SHARED_ LIB configuration item.
